azurerm/internal/services/datafactory/data_factory_linked_service_azure_databricks_resource.go

```diff
@@ -483,36 +454,29 @@ func resourceDataFactoryLinkedServiceDatabricksRead(d *schema.ResourceData, meta
 	d.Set("resource_group_name", resourceGroup)
 	d.Set("data_factory_name", dataFactoryName)
 
-	Databricks, ok := resp.Properties.AsAzureDatabricksLinkedService()
-
+	databricks, ok := resp.Properties.AsAzureDatabricksLinkedService()
 	if !ok {
-		return fmt.Errorf("Error classifiying Data Factory Linked Service Databricks %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", name, dataFactoryName, resourceGroup, datafactory.TypeAzureDatabricks, *resp.Type)
+		return fmt.Errorf("classifiying Data Factory Linked Service Databricks %q (Data Factory %q / Resource Group %q): Expected: %q Received: %q", name, dataFactoryName, resourceGroup, datafactory.TypeAzureDatabricks, *resp.Type)
 	}
 
 	// Check the properties and verify if authentication is set to MSI
-	if props := Databricks.AzureDatabricksLinkedServiceTypeProperties; props != nil {
+	if props := databricks.AzureDatabricksLinkedServiceTypeProperties; props != nil {
+		d.Set("adb_domain", props.Domain)
+
 		if props.Authentication != nil && props.Authentication == "MSI" {
-			authenticationMsi := make(map[string]interface{})
-			authenticationMsi["workspace_resource_id"] = props.WorkspaceResourceID
-			authenticationMsiArray := []interface{}{authenticationMsi}
-			d.Set("authentication_msi", authenticationMsiArray)
-		} else if props.AccessToken != nil {
-			// Check the data type of the access token so we know how to process it.
-			if accessToken := props.AccessToken; accessToken != nil {
-				// We only process AzureKeyVaultSecreReference because a string based access token is masked with asterisks in the GET response
-				// so we can't set it
-				if keyVaultPassword, ok := accessToken.AsAzureKeyVaultSecretReference(); ok {
-					if err := d.Set("authentication_key_vault_password", flattenAzureKeyVaultPassword(keyVaultPassword)); err != nil {
-						return fmt.Errorf("setting `authentication_key_vault_password`: %+v", err)
-					}
+			d.Set("msi_work_space_resource_id", props.WorkspaceResourceID)
+		} else if accessToken := props.AccessToken; accessToken != nil {
+			// We only process AzureKeyVaultSecreReference because a string based access token is masked with asterisks in the GET response
+			// so we can't set it
+			if keyVaultPassword, ok := accessToken.AsAzureKeyVaultSecretReference(); ok {
+				if err := d.Set("key_vault_password", flattenAzureKeyVaultPassword(keyVaultPassword)); err != nil {
+					return fmt.Errorf("setting `key_vault_password`: %+v", err)
 				}
 			}
 		}
 
-		if props.Domain != nil {
-			d.Set("adb_domain", props.Domain)
-		}
-
+		instancePoolArray := make([]interface{}, 0)
+		newClusterArray := make([]interface{}, 0)
```

Comment:
We always need to set a TypeList or a TypeSet to something even if it's empty. We can get some weird differences if we don't. So we'll declare an empty `[]interface` and set it outside of where we actually populate it

Reply:
Ah ok, good to know.